File: estructurales/facade/FacadeExample.py
class CPU:
    def freeze(self): print("CPU: freeze")
    def jump(self, position): print(f"CPU: jump to {position}")
    def execute(self): print("CPU: execute")

class Memory:
    def load(self, position, data): print(f"Memory: load '{data}' at {position}")

class Disk:
    def read(self, sector, size):
        return f"DATA(sector={sector}, size={size})"

class ComputerFacade:
    def __init__(self):
        self.cpu = CPU()
        self.memory = Memory()
        self.disk = Disk()

    def start(self):
        self.cpu.freeze()
        data = self.disk.read(10, 128)
        self.memory.load(0, data)
        self.cpu.jump(0)
        self.cpu.execute()

if __name__ == "__main__":
    computer = ComputerFacade()
    computer.start()

File: estructurales/facade/FacadeExample.java
class CPU {
    public void freeze() { System.out.println("CPU: freeze"); }
    public void jump(int position) { System.out.println("CPU: jump to " + position); }
    public void execute() { System.out.println("CPU: execute"); }
}

class Memory {
    public void load(int position, String data) {
        System.out.println("Memory: load '" + data + "' at " + position);
    }
}

class Disk {
    public String read(int sector, int size) {
        return "DATA(sector=" + sector + ", size=" + size + ")";
    }
}

class ComputerFacade {
    private final CPU cpu = new CPU();
    private final Memory memory = new Memory();
    private final Disk disk = new Disk();

    public void start() {
        cpu.freeze();
        String data = disk.read(10, 128);
        memory.load(0, data);
        cpu.jump(0);
        cpu.execute();
    }
}

public class FacadeExample {
    public static void main(String[] args) {
        ComputerFacade computer = new ComputerFacade();
        computer.start();
    }
}
